File: 프로그래머스/lv1/42748. K번째수/K번째수.js
function solution(array, commands) {
    const answer = [];
    
    for( let idx = 0; idx < commands.length; idx++ ) {
        const i = commands[idx][0]; // 이중배열의 첫번째값
        const j = commands[idx][1]; // 이중배열의 두번째값
        const k = commands[idx][2]; // 이중배열의 세번째값
        
        const result = array.slice( i - 1, j ) // 0부터 시작(-1)
                            .sort( (a, b) => {
                                return a - b //오름차순
                            })
        answer.push( result[k - 1] ) //인덱스 0부터 시작 
    }
    
    return answer;
}

File: 프로그래머스/lv1/42748. K번째수/K번째수.py
# def solution(array, commands):
#     answer = []
#     for i in range(len(commands)):
#         start = commands[i][0]
#         stop = commands[i][1]
#         target = commands[i][2]
#         result = array[start-1:stop]
#         result.sort()
#         for j in range(len(result)):
#             if j == target-1:
#                 answer.append(result[j])
#     return answer
# def solution(array, commands):
#     answer = []
#     for i in range(len(commands)):
#         start = commands[i][0]
#         stop = commands[i][1]
#         target = commands[i][2]
#         result = array[start-1:stop]
#         result.sort()
#         for j in range(len(result)):
#             if j == target-1:
#                 answer.append(result[j])
#     return answer

def solution(array,commands):
    answer = []

    for i in commands:
        ary = array[i[0]-1 : i[1]] #문제에서 주어진 크기만큼 자르기

        ary.sort() # 오름차순
        answer.append(ary[i[2]-1]) # k번째 값 넣기 

    return answer